test/integration/dynamic-routing/test/index.test.js

```diff
@@ -1062,6 +1067,10 @@ function runTests({ dev, serverless }) {
     })
 
     it('should show error when interpolating fails for href', async () => {
+      if (await fs.pathExists(join(appDir, 'middleware.js'))) {
+        return
+      }
```

Comment:
**suggestion:** maybe we could remove the middleware file at the beginning of this test, and use `beforeEach()` to write the middleware content (so all other tests have it).

Right now, we don't really know when the test is used or not, this it'll be silently green, pretending it's working.

Reply:
Updated in https://github.com/vercel/next.js/pull/37784/commits/1ea3288386ff0b1898e73084ccda64b64cac15ec